go/src/koding/kites/kloud/utils/object/printer.go

```diff
@@ -0,0 +1,113 @@
+package object
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/fatih/structs"
+)
+
+type Printer struct {
+	Tag    string
+	JSON   bool
+	Format string
+	W      io.Writer
+}
+
+func (p *Printer) Print(v interface{}) error {
+	if p.JSON {
+		return p.printJSON(v)
+	}
+	return p.printTab(v)
+}
+
+func (p *Printer) printJSON(v interface{}) error {
+	enc := json.NewEncoder(p.W)
+	enc.SetIndent("", "\t")
+
+	return enc.Encode(v)
+}
+
+var (
+	fprintf  = reflect.ValueOf(fmt.Fprintf)
+	stringer = reflect.TypeOf((*fmt.Stringer)(nil)).Elem()
+)
+
+func (p *Printer) printTab(v interface{}) error {
```

Comment:
@cihangir I think about showing headers(keys) when th array is empty. Example from docker:
```
(archpk) {machine_mount} ~/koding/koding/go/src/koding$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```

The same behavior is in `kd (?:machine)? list` (I'm not able to show it because my klient is currently down -.-)